src/kaleidoscope_internal/storage.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#define _INLINE_STORAGE_HANDLER_FOR_PLUGIN(PLUGIN)                                       __NL__ \
+  EventHandler__::call(PLUGIN, current_offset, hook_args...);                            __NL__ \
+  current_offset += PLUGIN.storageSize();
+
+#define _INLINE_STORAGE_OFFSET_COUNTER(PLUGIN)                                           __NL__ \
+  if ((uint16_t)&PLUGIN == (uint16_t)&plugin)                                            __NL__ \
+    return current_offset;                                                               __NL__ \
+  current_offset += PLUGIN.storageSize();
```

Comment:
Orderwise, I feel like maybe the single-use inlines should be just before the thing that uses them, maybe?

Reply:
Good point, will move them. Thanks!

Reply:
On a second look: they are kinda just before the thing that uses them. `_KALEIDOSCOPE_INIT_STORAGE_PLUGINS` is the only user, and these two are just above that. I could split it, and move one of the `_INLINE_` things in between, but I don't think that'd make things any clearer.